monGARS/core/dynamic_response.py

```
@@ -134,6 +134,63 @@ async def get_personality_traits(
+            async with self._lock:
+                self._pending.pop(cache_key, None)
+                if self._cache_ttl < 0 and self._permanent_cache is not None:
+                    self._permanent_cache.pop(cache_key, None)
+                if self._ttl_cache is not None:
+                    self._ttl_cache.pop(cache_key, None)
+
+        try:
+            traits = await self.get_personality_traits(user_id, interactions_list)
+        except Exception as exc:  # pragma: no cover - defensive logging
+            logger.exception(
+                "Failed to estimate personality for %s: %s", user_id, exc
```

Ruff (0.14.0)

Redundant exception object included in `logging.exception` call

(TRY401)